From cb4569ad779d0bea1758c49e8aa476960bd45024 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinh.nguyen@intel.com>
Date: Thu, 8 Jul 2021 16:02:45 -0500
Subject: [PATCH 3/9] HSD #14012818438-2: dts: Add hwmon support for temp and
 voltage sensing

commit 3d09488d859076d04c1bb8a6fb60ea60c889e84e from
https://github.com/altera-opensource/linux-socfpga/commits/socfpga-5.4.124-lts

Add hwmon support for temperature and voltage sensing to the Agilex/N5X
platform.

Signed-off-by: Kris Chaplin <kris.chaplin@intel.com>
Signed-off-by: Dinh Nguyen <dinh.nguyen@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 arch/arm64/boot/dts/intel/socfpga_agilex.dtsi |  4 ++
 .../boot/dts/intel/socfpga_n5x_socdk.dts      | 46 +++++++++++++++++++
 2 files changed, 50 insertions(+)

diff --git a/arch/arm64/boot/dts/intel/socfpga_agilex.dtsi b/arch/arm64/boot/dts/intel/socfpga_agilex.dtsi
index 4323bf75b864..a9a9fb789d19 100644
--- a/arch/arm64/boot/dts/intel/socfpga_agilex.dtsi
+++ b/arch/arm64/boot/dts/intel/socfpga_agilex.dtsi
@@ -618,6 +618,10 @@
 		};
 
 		firmware {
+			temp_volt: hwmon {
+				compatible = "intel,soc64-hwmon";
+			};
+
 			svc {
 				compatible = "intel,agilex-svc";
 				method = "smc";
diff --git a/arch/arm64/boot/dts/intel/socfpga_n5x_socdk.dts b/arch/arm64/boot/dts/intel/socfpga_n5x_socdk.dts
index b6152f312cc7..bdf56f575364 100644
--- a/arch/arm64/boot/dts/intel/socfpga_n5x_socdk.dts
+++ b/arch/arm64/boot/dts/intel/socfpga_n5x_socdk.dts
@@ -86,3 +86,49 @@
 &watchdog0 {
 	status = "okay";
 };
+
+&temp_volt {
+	voltage {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		input@2 {
+			label = "0.8V VDD";
+			reg = <2>;
+		};
+
+		input@3 {
+			label = "0.8V VDD_SDM";
+			reg = <3>;
+		};
+
+		input@4 {
+			label = "1.8V VCCADC";
+			reg = <4>;
+		};
+
+		input@5 {
+			label = "1.8V VCCPD";
+			reg = <5>;
+		};
+
+		input@6 {
+			label = "1.8V VCCIO_SDM";
+			reg = <6>;
+		};
+
+		input@7 {
+			label = "0.8V VDD_HPS";
+			reg = <7>;
+		};
+	};
+
+	temperature {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		input@1 {
+			label = "Main Die";
+			reg = <0>;
+		};
+	};
+};
-- 
2.31.1

